client/dashboard/sites/overview/index.tsx

```diff
@@ -70,6 +78,41 @@ function getGridLayout( {
 	};
 }
 
+function useSecondaryUpsellStack( site: Site ) {
+	const { data: domains = [], isLoading } = useQuery( {
+		...siteDomainsQuery( site.ID ),
+		select: ( data ) => {
+			// If the site has *.wpcomstaging.com domain, exclude *.wordpress.com
+			if ( data && data.find( ( domain ) => domain.is_wpcom_staging_domain ) ) {
+				return data.filter( ( domain ) => ! domain.wpcom_domain || domain.is_wpcom_staging_domain );
+			}
+
+			return data;
+		},
+		enabled: ! site.is_wpcom_staging_site,
+	} );
+
+	const shouldShowDIFMUpsellCard = getShouldShowDIFMUpsellCard( site );
+	const shouldShowDomainTransferUpsellCard = getShouldShowDomainTransferUpsellCard( site, domains );
+	const shouldShowDomainUpsellCard =
+		! shouldShowDomainTransferUpsellCard && getShouldShowDomainUpsellCard( domains );
+
+	const isEligibleSecondaryUpsellStack = ! site.is_wpcom_staging_site;
+	const shouldShowSecondaryUpsellStack =
+		isEligibleSecondaryUpsellStack &&
+		( shouldShowDIFMUpsellCard ||
+			shouldShowDomainTransferUpsellCard ||
+			shouldShowDomainUpsellCard );
+
+	return {
+		shouldShowDIFMUpsellCard,
+		shouldShowDomainTransferUpsellCard,
+		shouldShowDomainUpsellCard,
+		shouldShowSecondaryUpsellStack,
+		isLoading,
+	};
```

Comment:
I guess we can simply return `cards` so we don't need to check these logics again on the parent